test/app.e2e-spec.ts

```diff
@@ -11,16 +11,43 @@ import { InjectModel, TypegooseModule, getModelToken } from '../src';
 import { prop } from '@typegoose/typegoose';
 import { MongoMemoryServer } from 'mongodb-memory-server';
 
-const mongod = new MongoMemoryServer();
+let mongod: MongoMemoryServer;
+let mongoUri: string;
+let app: INestApplication;
+
+beforeAll(async () => {
+
+  mongod = await MongoMemoryServer.create();
+
+  mongoUri = mongod.getUri();
+
+  const moduleFixture = await Test.createTestingModule({
+    imports: [MockApp, MockSubModule]
+  }).compile();
+
+  app = moduleFixture.createNestApplication();
+  await app.init();
+});
+
+afterAll(() => mongod.stop());
 
 @Module({
-  imports: [TypegooseModule.forRoot('mongoose:uri')]
+  imports: [ TypegooseModule.forRootAsync({
+    useFactory: () => {
+      return {
+        uri: mongoUri,
```

Comment:
The test runner (travis ci) seems to be dead so I do not know if the tests are running. But doesn't `uri: mongoUri` always evaluate to `undefined` since the URI is created in the `beforeAll` and the class is not (created during compile).

Reply:
It's why I changed it to `forRootAsync`. That way, the URI is available during initialization.

Scott

Reply:
Ahh, yes :)